rclcpp/src/rclcpp/intra_process_manager.cpp

```diff
@@ -141,9 +140,10 @@ IntraProcessManager::get_subscription_count(uint64_t intra_process_publisher_id)
     return 0;
   }
 
-  auto count =
-    publisher_it->second.take_shared_subscriptions.size() +
-    publisher_it->second.take_ownership_subscriptions.size();
+  auto count = 0u;
+  for (auto i = 0u; i < SplittedSubscriptions::IndexNum; ++i) {
```

Comment:
you can use [std::accumulate](https://en.cppreference.com/w/cpp/algorithm/accumulate) if you want, though there's no big benefit on using it.